packages/desktop-client/src/components/autocomplete/CategoryAutocomplete.tsx

```diff
@@ -183,6 +199,34 @@ export function CategoryAutocomplete({
     [defaultCategoryGroups, categoryGroups, showSplitOption],
   );
 
+  const filterSuggestions = useCallback(
+    (
+      suggestions: CategoryAutocompleteItem[],
+      value: string,
+    ): CategoryAutocompleteItem[] => {
+      return suggestions
+        .filter(suggestion => {
+          const suggestionAsItem = suggestion as CategoryAutocompleteItem; // No need for casting to Suggestion here
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏no need for this type coercion anymore
```suggestion
        .filter(suggestion => {
```